--- e2fsprogs-1.45.4/lib/et/Makefile.in	2019-09-23 22:04:59.000000000 +0200
+++ e2fsprogs-1.45.4/lib/et/Makefile.in	2019-11-18 10:35:48.505299801 +0100
@@ -101,10 +101,13 @@
 
 installdirs::
 	$(E) "	MKDIR_P $(libdir) $(includedir)/et $(datadir)/et $(bindir) $(man1dir) $(man3dir)"
-	$(Q) $(MKDIR_P) $(DESTDIR)$(libdir) \
-		$(DESTDIR)$(includedir)/et $(DESTDIR)$(datadir)/et \
-		$(DESTDIR)$(bindir) $(DESTDIR)$(man1dir) \
-		$(DESTDIR)$(man3dir) $(DESTDIR)$(pkgconfigdir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(libdir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(includedir)/et
+	$(Q) $(MKDIR_P) $(DESTDIR)$(datadir)/et
+	$(Q) $(MKDIR_P) $(DESTDIR)$(bindir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(man1dir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(man3dir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(pkgconfigdir)
 
 install:: compile_et libcom_err.a $(HFILES) installdirs com_err.pc
 	$(E) "	INSTALL_DATA $(libdir)/libcom_err.a"
--- e2fsprogs-1.45.4/lib/ss/Makefile.in	2019-09-23 22:04:59.000000000 +0200
+++ e2fsprogs-1.45.4/lib/ss/Makefile.in	2019-11-18 10:44:39.300421836 +0100
@@ -125,9 +125,11 @@
 
 installdirs::
 	$(E) "	MKDIR_P $(libdir) $(includedir)/ss $(datadir)/ss $(bindir)"
-	$(Q) $(MKDIR_P) $(DESTDIR)$(libdir) \
-		$(DESTDIR)$(includedir)/ss $(DESTDIR)$(datadir)/ss \
-		$(DESTDIR)$(bindir) $(DESTDIR)$(pkgconfigdir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(libdir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(includedir)/ss
+	$(Q) $(MKDIR_P) $(DESTDIR)$(datadir)/ss
+	$(Q) $(MKDIR_P) $(DESTDIR)$(bindir)
+	$(Q) $(MKDIR_P) $(DESTDIR)$(pkgconfigdir)
 
 install:: libss.a $(INSTALL_HFILES) installdirs ss_err.h mk_cmds ss.pc
 	$(E) "	INSTALL_DATA $(DESTDIR)$(libdir)/libss.a"
