--- kwin-5.12.5/plugins/kdecorations/aurorae/themes/plastik/package/contents/ui/main.qml
+++ kwin-5.12.5/plugins/kdecorations/aurorae/themes/plastik/package/contents/ui/main.qml
@@ -77,9 +77,8 @@
         }
         root.animateButtons = decoration.readConfig("animateButtons", true);
         if (decoration.animationsSupported) {
             root.animationDuration = 150;
-            root.animateButtons = false;
         }
     }
     ColorHelper {
         id: colorHelper
--- kwin-5.12.5/plugins/kdecorations/aurorae/themes/plastik/package/contents/ui/PlastikButton.qml
+++ kwin-5.12.5/plugins/kdecorations/aurorae/themes/plastik/package/contents/ui/PlastikButton.qml
@@ -21,6 +21,10 @@
 DecorationButton {
     id: button
     function colorize() {
+        animConturTop.duration = root.animateButtons && !button.pressed ? root.animationDuration : 0;
+        animConturBottom.duration = root.animateButtons && !button.pressed ? root.animationDuration : 0;
+        animSurfaceTop.duration = root.animateButtons && !button.pressed ? root.animationDuration : 0;
+        animSurfaceBottom.duration = root.animateButtons && !button.pressed ? root.animationDuration : 0;
         var highlightColor = null;
         if (button.pressed) {
             if (button.buttonType == DecorationOptions.DecorationButtonClose) {
@@ -62,16 +66,16 @@
     property color baseSurfaceTop: colorHelper.shade(options.titleBarColor, ColorHelper.MidlightShade, colorHelper.contrast - 0.4)
     property color baseSurfaceBottom: colorHelper.shade(options.titleBarColor, ColorHelper.LightShade, colorHelper.contrast - 0.4)
     Behavior on conturTop {
-        ColorAnimation { duration: root.animateButtons ? root.animationDuration : 0 }
+        ColorAnimation { id: animConturTop; duration: root.animateButtons ? root.animationDuration : 0 }
     }
     Behavior on conturBottom {
-        ColorAnimation { duration: root.animateButtons ? root.animationDuration : 0 }
+        ColorAnimation { id: animConturBottom; duration: root.animateButtons ? root.animationDuration : 0 }
     }
     Behavior on surfaceTop {
-        ColorAnimation { duration: root.animateButtons ? root.animationDuration : 0 }
+        ColorAnimation { id: animSurfaceTop; duration: root.animateButtons ? root.animationDuration : 0 }
     }
     Behavior on surfaceBottom {
-        ColorAnimation { duration: root.animateButtons ? root.animationDuration : 0 }
+        ColorAnimation { id: animSurfaceBottom; duration: root.animateButtons ? root.animationDuration : 0 }
     }
     width: size
     height: size
