--- libvpx-1.8.1/configure	2019-07-15 23:55:33.000000000 +0200
+++ libvpx-1.8.1/configure	2020-02-01 13:44:38.754820409 +0100
@@ -165,10 +165,6 @@
     [ -f "${source_path}/${t}.mk" ] && enable_feature ${t}
 done
 
-if ! diff --version >/dev/null; then
-  die "diff missing: Try installing diffutils via your package manager."
-fi
-
 if ! perl --version >/dev/null; then
     die "Perl is required to build"
 fi
